src/upgrades/v0_9_0/ca_objects_migration.rs

```diff
@@ -145,109 +146,102 @@ impl UpgradeStore for CasStoreMigration {
+                            let old_cmd: OldStoredCaCommand = self.get(first_command)?;
+                            old_cmd.time
+                        }
+                        None => Time::now(),
+                    };
 
-            // Find all command keys and sort them by sequence.
-            // Then turn them back into key store keys for further processing.
-            let cmd_keys = self.command_keys(&migration_scope)?;
+                    let repo_response = self.repo_manager.repository_response(&id)?;
 
-            // Get the time to use if we need to inject commands and events for the init
-            // event, based on the first recorded time in command keys.
-            let time_for_init_command = match cmd_keys.first() {
-                Some(first_command) => {
-                    let old_cmd: OldStoredCaCommand = self.get(first_command)?;
-                    old_cmd.time
-                }
-                None => Time::now(),
-            };
-
-            // Migrate the init event.
-            let old_init_key = KeyStoreKey::scoped(migration_scope.clone(), "delta-0.json".to_string());
-            let init_key = KeyStoreKey::scoped(scope.clone(), "delta-0.json".to_string());
-
-            let old_init: OldCaIni = self.get(&old_init_key)?;
-            let (id, _, old_ini_det) = old_init.unpack();
-            let (rfc_8183_id, repo_opt, ta_opt) = old_ini_det.unpack();
-            let ini = IniDet::new(&id, rfc_8183_id.into());
-            debug!("  init: {}", init_key);
-            self.store.store(&init_key, &ini)?;
-
-            // If the CA was initialized with an embedded repository, then make sure that
-            // we generate a command + events to update the repository to the 'local' RFC 8181
-            // version of this repository.
-
-            if repo_opt.is_some() {
-                debug!("  +- Generate command and event for initialized repository");
-                let res = self.repo_manager.repository_response(&id)?;
-
-                let contact = RepositoryContact::new(res);
-                let service_uri = contact.service_uri().clone();
-
-                info.last_event += 1;
-                let event = CaEvt::new(&id, info.last_event, CaEvtDet::RepoUpdated { contact });
-
-                let command = StoredCaCommand::new(
-                    dflt_actor.clone(),
-                    time_for_init_command,
-                    id,
-                    info.last_event,
-                    info.last_command,
-                    StorableCaCommand::RepoUpdate { service_uri },
-                    StoredEffect::Success { events: vec![1] },
-                );
+                    let contact = RepositoryContact::new(repo_response);
+                    let service_uri = contact.service_uri().clone();
 
-                info.last_command += 1;
-                info.last_update = time_for_init_command;
+                    data_upgrade_info.last_event += 1;
```

Comment:
In the old code you incremented this AFTER creating the stored CA command, now you do it before. Was there an off by one error then?

Reply:
no, but it was all subtly different because I was updating another file in place. Here I am tracking how far the migration got and the `info` is only updated at the end.